sdk/servicebus/service-bus/src/old/oldServiceBusReceiverClient.ts

```diff
@@ -0,0 +1,358 @@
+  OnError,
+  MessageHandlerOptions,
+  CorrelationFilter,
+  RuleDescription
+} from "..";
+import { isTokenCredential } from "@azure/core-amqp";
+import { ClientEntityContext } from "../clientEntityContext";
+import { ClientType } from "../client";
+import { ReceiveMode, ServiceBusMessage, ReceivedMessageInfo } from "../serviceBusMessage";
+import { SessionReceiverOptions } from "../session/messageSession";
+import { generate_uuid } from "rhea-promise";
+import { ConnectionContext } from "../connectionContext";
+
+export type ServiceBusClientReceiverOptions = ServiceBusClientOptions & SessionReceiverOptions;
+
+/**
+ * @internal
+ * @ignore
+ */
+export class ServiceBusReceiverClient {
+  public _receiveMode: ReceiveMode;
+  public _entityPath: string;
+  private _currentReceiver: InternalReceiver | InternalSessionReceiver;
+  readonly defaultRuleName: string = "$Default";
+
+  // Queue
+  constructor(
+    entityConnectionString: string,
+    receiveMode?: ReceiveMode,
+    options?: ServiceBusClientReceiverOptions
+  );
+
+  // Queue
+  constructor(
+    serviceBusConnectionString: string,
+    entityName: string,
+    receiveMode?: ReceiveMode,
+    options?: ServiceBusClientReceiverOptions
+  );
+
+  // Queue
+  constructor(
+    host: string,
+    entityName: string,
+    credential: TokenCredential,
+    receiveMode?: ReceiveMode,
+    options?: ServiceBusClientReceiverOptions
+  );
+
+  // Subscription
+  constructor(
+    topicConnectionString: string,
+    subscriptionName: string,
+    receiveMode?: ReceiveMode,
+    options?: ServiceBusClientReceiverOptions
+  );
+
+  // Subscription
+  constructor(
+    serviceBusConnectionString: string,
+    topicName: string,
+    subscriptionName: string,
+    receiveMode?: ReceiveMode,
+    options?: ServiceBusClientReceiverOptions
+  );
+
+  // Subscription
+  constructor(
+    host: string,
+    topicName: string,
+    subscriptionName: string,
+    credential: TokenCredential,
+    receiveMode?: ReceiveMode,
+    options?: ServiceBusClientReceiverOptions
+  );
+
+  constructor(
+    param1: string,
+    param2?: string | ReceiveMode,
+    param3?: string | ReceiveMode | ServiceBusClientReceiverOptions | TokenCredential,
+    param4?: ServiceBusClientReceiverOptions | ReceiveMode | TokenCredential,
+    param5?: ServiceBusClientReceiverOptions | ReceiveMode,
+    param6?: ServiceBusClientReceiverOptions
+  ) {
+    let receiveMode: ReceiveMode;
+    let options: ServiceBusClientReceiverOptions;
+    let context: ConnectionContext;
+    if (typeof param2 !== "string") {
+      // Queue
+      // (entityConnectionString: string, receiveMode: ReceiveMode, options?: ServiceBusClientReceiverOptions)
+      const entityConnectionString = param1;
+      options = param3 as ServiceBusClientReceiverOptions;
+      // get the entity name from the connection string
+      const entityPathMatch = entityConnectionString.match(/^.+EntityPath=(.+?);{0,1}$/);
+
+      if (entityPathMatch!.length !== 2) {
```

Comment:
Can't entityPathMatch be null if no matches are found? Then you'd get an error trying to access the `length` property.

Reply:
_yes_. This is the old client so you can more or less ignore it. The version I moved over into constructorHelpers does not have this bug (it checks for entityPathMatch != null first).

Reply:
Oh gotcha.